pkg/ovsdb/ovsdb.go

```diff
@@ -0,0 +1,450 @@
+// Not in use
+func (self *OvsDriver) CreateBridge(bridgeName string, failMode string, stp bool) error {
+	namedUuidStr := "dummy"
+	protocols := []string{"OpenFlow10", "OpenFlow11", "OpenFlow12", "OpenFlow13"}
+
+	// If the bridge already exists, just return
+	// FIXME: should we delete the old bridge and create new one?
+	if self.IsBridgePresent(bridgeName) {
+		return nil
+	}
+
+	// simple insert/delete operation
+	brOp := libovsdb.Operation{}
+	bridge := make(map[string]interface{})
+	bridge["name"] = bridgeName
+	bridge["protocols"], _ = libovsdb.NewOvsSet(protocols)
+	bridge["stp_enable"] = stp
+	brOp = libovsdb.Operation{
+		Op:       "insert",
+		Table:    "Bridge",
+		Row:      bridge,
+		UUIDName: namedUuidStr,
+	}
+
+	// Inserting/Deleting a Bridge row in Bridge table requires mutating
+	// the open_vswitch table.
+	brUuid := []libovsdb.UUID{{GoUUID: namedUuidStr}}
+	mutateUuid := brUuid
+	mutateSet, _ := libovsdb.NewOvsSet(mutateUuid)
+	mutation := libovsdb.NewMutation("bridges", "insert", mutateSet)
+	condition := libovsdb.NewCondition("_uuid", "==", self.getRootUuid())
+
+	// simple mutate operation
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     "Open_vSwitch",
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	operations := []libovsdb.Operation{brOp, mutateOp}
+
+	// operations := []libovsdb.Operation{brOp}
+	return self.ovsdbTransact(operations)
+}
+
+// Delete a bridge from ovs
+// Not in use
+func (self *OvsDriver) DeleteBridge(bridgeName string) error {
+	// lock the cache for read
+	self.lock.RLock()
+	defer self.lock.RUnlock()
+
+	namedUuidStr := "dummy"
+	brUuid := []libovsdb.UUID{{GoUUID: namedUuidStr}}
+
+	// simple insert/delete operation
+	brOp := libovsdb.Operation{}
+	condition := libovsdb.NewCondition("name", "==", bridgeName)
+	brOp = libovsdb.Operation{
+		Op:    "delete",
+		Table: "Bridge",
+		Where: []interface{}{condition},
+	}
+
+	// also fetch the br-uuid from cache
+	for uuid, row := range self.ovsdbCache["Bridge"] {
+		name := row.Fields["name"].(string)
+		if name == bridgeName {
+			brUuid = []libovsdb.UUID{{GoUUID: uuid}}
+			break
+		}
+	}
+
+	// Inserting/Deleting a Bridge row in Bridge table requires mutating
+	// the open_vswitch table.
+	mutateUuid := brUuid
+	mutateSet, _ := libovsdb.NewOvsSet(mutateUuid)
+	mutation := libovsdb.NewMutation("bridges", "delete", mutateSet)
+	condition = libovsdb.NewCondition("_uuid", "==", self.getRootUuid())
+
+	// simple mutate operation
+	mutateOp := libovsdb.Operation{
+		Op:        "mutate",
+		Table:     "Open_vSwitch",
+		Mutations: []interface{}{mutation},
+		Where:     []interface{}{condition},
+	}
+
+	operations := []libovsdb.Operation{brOp, mutateOp}
+	return self.ovsdbTransact(operations)
+}
+
+// Create an internal port in OVS
+func (self *OvsDriver) CreatePort(intfName, intfType string, vlanTag uint, contIfaceName, contNetnsPath, ovnPortName string) error {
```

Comment:
Could you make it so we pass contIfaceName, contNetnsPath and ovnPortName as a arbitrary map? This is hard to read when it is called and is too specific.